addons/udes_stock/models/stock_move.py

```diff
@@ -519,3 +519,53 @@ def refactor_by_move_line_groups(self, groups):
             })
 
         return result_moves
+
+    @api.multi
+    def refactor_action_batch_pickings_by_date_priority(self):
+        """Batch pickings by date and priority."""
+        self._refactor_action_batch_pickings_by(
+            lambda picking: (picking.scheduled_date, picking.priority)
+        )
+
+    @api.multi
+    def refactor_action_batch_pickings_by_date(self):
+        """Batch pickings by date."""
+        self._refactor_action_batch_pickings_by(
+            lambda picking: (picking.scheduled_date,)
+        )
+
+    def _refactor_action_batch_pickings_by(self, by_key):
+        """Group picks in batches.
+
+        Args:
+            by_key (function): The function to generate the key to group pickings by.
+                Should return a :obj:`tuple`.
+        """
+        PickingBatch = self.env["stock.picking.batch"]
+
+        # Precache pickings.
+        self.mapped("picking_id")
+
+        # Find existing draft batches.
+        batches = PickingBatch.search([("state", "=", "draft")])
```

Comment:
Also filter by picking type. All the pickings in self should be of one picking type.

Reply:
The reason is because we can have the same refactoring applied to different picking types, and we don't want the pickings of different picking types to be merged into the same batch.

Reply:
```python
    ("picking_type", "=", self.picking_type)
```
?

Reply:
Or 
```python
    (self.picking_type_id, "in", "picking_type_ids")
```
?

Reply:
`('picking_type_ids', 'in', picking_types.ids)`

Reply:
`picking_type` isn't a variable - do you mean the `self.picking_type_id`?

Reply:
Yes, something like:
`picking_types = self.mapped('picking_type_id')`
Because self can contain more than one stock move.